daemon/volumes.go

```diff
@@ -217,7 +217,7 @@ func (daemon *Daemon) registerMountPoints(container *container.Container, hostCo
 			}
 		}
 
-		if mp.Type == mounttypes.TypeBind {
+		if mp.Type == mounttypes.TypeBind && (cfg.BindOptions == nil || !cfg.BindOptions.CreateMountpoint) {
```

Comment:
We'll probably need to gate the option by API version, and (probably) produce an error when trying to use it with a "non bind-mount" mount.

Reply:
> produce an error when trying to use it with a "non bind-mount" mount
the API defines `BindOptions`/`VolumeOptions`/`TmpfsOptions` and engine ignores those for non-matching type. Do you suggest we should reject an API call then (which would be a potential breaking change)

Reply:
Good one; we should've done that in the first place (not accept `BindOptions` on a `type=Volume` etc); I guess "in general" we passed that ship. We could still fix it for _this_ option 🤔 

@cpuguy83 @tianon - any thoughts? (silently ignore the invalid option, or error?)

Reply:
-1 on adding a check _just_ for this option, as this would make the code pretty odd to read. Better we check those options only are set for matching type if client uses API >= 1.42

Reply:
Any way we could survey some popular API users and see how heavy the breakage of rejecting what are frankly nonsensical requests might be?  IMO that's a breaking change that actually does make some sense to make, since the requests being made to the daemon in those cases don't actually make any sense and aren't something we could successfully fulfill (even if we didn't explicitly error out previously).

Reply:
^^ a commit was added to invalidate the request on any of these options (thanks!)